feat: monitoring stack tutorial (#33)
AshAvalanche/ash-docs: docs/toolkit/ansible-avalanche-collection/tutorials/monitoring.md

---
sidebar_position: 9
---

> # Monitoring

Reviewer: ```suggestion
# Nodes monitoring
```

> In this section, we will learn how to use the `ash.avalanche.install_monitoring_stack` playbook to deploy a monitoring stack for Avalanche.

Reviewer: ```suggestion
In this section, we will learn how to use the `ash.avalanche.install_monitoring_stack` playbook to deploy a monitoring stack for Avalanche nodes.
```

The monitoring stack is comprised of:

- [Prometheus](https://prometheus.io)
- [Prometheus Node Exporter](https://prometheus.io/docs/guides/node-exporter/)
- [Grafana](https://grafana.com)

:::note
For this tutorial, we will use the [`local`](https://github.com/AshAvalanche/ansible-avalanche-getting-started/tree/main/inventories/local) inventory provided by the [Ansible Avalanche Getting Started](https://github.com/AshAvalanche/ansible-avalanche-getting-started) repository. See [Create a local test network](./local-test-network) for details.
:::

## Requirements

The `ash.avalanche.install_monitoring_stack` playbook installs Prometheus and Grafana through the [prometheus.prometheus](https://galaxy.ansible.com/prometheus/prometheus) collection and the [cloudalchemy.grafana](https://galaxy.ansible.com/cloudalchemy/grafana) role.

We can make these available locally with:

```bash
ansible-galaxy install -r ansible_collections/ash/avalanche/requirements.yml
```

## Install the monitoring stack

> Grafana will be installed on the node targeted by the `grafana` Ansible group and Prometheus on the `prometheus` Ansible group. In [Ansible Avalanche Getting Started](https://github.com/AshAvalanche/ansible-avalanche-getting-started), it is `frontend` by default. You can change this by editing the [`hosts`](https://github.com/AshAvalanche/ansible-avalanche-getting-started/blob/main/inventories/local/hosts) file.

Reviewer: ```suggestion
Grafana will be installed on the nodes of the `grafana` Ansible group and Prometheus on the `prometheus` Ansible group. In [Ansible Avalanche Getting Started](https://github.com/AshAvalanche/ansible-avalanche-getting-started), it is the `frontend` node by default. You can change this by editing the [`hosts`](https://github.com/AshAvalanche/ansible-avalanche-getting-started/blob/main/inventories/local/hosts) file.
```
